timelock-impl/src/test/java/com/palantir/atlasdb/timelock/paxos/PaxosTimestampBoundStoreTest.java

```diff
@@ -315,61 +249,161 @@ public void canGetAgreedStateAfterNodeDown() {
+    public void canForceAgreedState(boolean useBatch) {
+        setup(useBatch);
         assertThat(store.forceAgreedState(1, FORTY_TWO)).isEqualTo(ONE_AND_FORTY_TWO);
         assertThat(store.getAgreedState(1)).isEqualTo(ONE_AND_FORTY_TWO);
     }
 
-    @Test
-    public void forceAgreedStateCanBeUsedToGainKnowledge() {
+    @ParameterizedTest
+    @MethodSource("getParameters")
+    public void forceAgreedStateCanBeUsedToGainKnowledge(boolean useBatch) {
+        setup(useBatch);
         assertThat(store.forceAgreedState(1, FORTY_TWO)).isEqualTo(ONE_AND_FORTY_TWO);
 
         PaxosTimestampBoundStore additionalStore = createPaxosTimestampBoundStore(1);
         assertThat(additionalStore.forceAgreedState(1, null)).isEqualTo(ONE_AND_FORTY_TWO);
     }
 
-    @Test
-    public void forceAgreedStateReturnsFirstForcedValue() {
+    @ParameterizedTest
+    @MethodSource("getParameters")
+    public void forceAgreedStateReturnsFirstForcedValue(boolean useBatch) {
+        setup(useBatch);
         assertThat(store.forceAgreedState(1, FORTY_TWO)).isEqualTo(ONE_AND_FORTY_TWO);
         assertThat(store.forceAgreedState(1, 1L)).isEqualTo(ONE_AND_FORTY_TWO);
         assertThat(store.getAgreedState(1)).isEqualTo(ONE_AND_FORTY_TWO);
     }
 
-    @Test
-    public void forceAgreedStateOperatesAtSequenceNumberLevel() {
+    @ParameterizedTest
+    @MethodSource("getParameters")
+    public void forceAgreedStateOperatesAtSequenceNumberLevel(boolean useBatch) {
+        setup(useBatch);
         long fortyThree = FORTY_TWO + 1;
         assertThat(store.forceAgreedState(1, FORTY_TWO)).isEqualTo(ONE_AND_FORTY_TWO);
         assertThat(store.forceAgreedState(0, fortyThree)).isEqualTo(ImmutableSequenceAndBound.of(0, fortyThree));
         assertThat(store.getAgreedState(0)).isEqualTo(ImmutableSequenceAndBound.of(0, fortyThree));
         assertThat(store.getAgreedState(1)).isEqualTo(ONE_AND_FORTY_TWO);
     }
 
-    @Test
-    public void forceAgreedStateThrowsIfNoStateWasAgreedUpon() {
+    @ParameterizedTest
+    @MethodSource("getParameters")
+    public void forceAgreedStateThrowsIfNoStateWasAgreedUpon(boolean useBatch) {
+        setup(useBatch);
         assertThatThrownBy(() -> store.forceAgreedState(1, null)).isInstanceOf(NullPointerException.class);
     }
 
-    @Test
-    public void retriesForceAgreedStateUntilSuccessful() throws Exception {
+    @ParameterizedTest
+    @MethodSource("getParameters")
+    public void retriesForceAgreedStateUntilSuccessful(boolean useBatch) throws Exception {
+        setup(useBatch);
         PaxosProposer wrapper = spy(new OnceFailingPaxosProposer(createPaxosProposer(0)));
         store = createPaxosTimestampBoundStore(0, wrapper);
         store.forceAgreedState(1, TIMESTAMP_1);
         assertThat(store.getUpperLimit()).isGreaterThanOrEqualTo(TIMESTAMP_1);
         verify(wrapper, times(2)).propose(anyLong(), any(byte[].class));
     }
 
+    public void setup(boolean useBatch) {
```

Comment:
Is it worth extracting the common part of the function to a `@BeforeEach` function? And only leaving the part that depends on the parameterized `useBatch` here?